diff -Naru awesome-4.3/awesomerc.lua awesome-4.3-patched/awesomerc.lua
--- awesome-4.3/awesomerc.lua	2019-01-28 01:53:24.000000000 -0500
+++ awesome-4.3-patched/awesomerc.lua	2024-03-01 02:44:49.433213944 -0500
@@ -19,6 +19,9 @@
 -- when client with a matching name is opened:
 require("awful.hotkeys_popup.keys")
 
+-- Enable compositor, helps with screen tearing
+awful.util.spawn_with_shell("xcompmgr &")
+
 -- {{{ Error handling
 -- @DOC_ERROR_HANDLING@
 -- Check if awesome encountered an error during startup and fell back to
